Controller/ResourceController.php

```diff
@@ -123,17 +123,38 @@ protected function findOrThrowNotFound(ResourceInterface $resource, $id)
     protected function getCollectionData(ResourceInterface $resource, Request $request)
     {
         $page = (int) $request->get('page', 1);
-
+        $order = [];
         $filters = [];
-        foreach ($resource->getFilters() as $resourceFilter) {
+        $resourceFilters = $resource->getFilters();
+
+        foreach ($resourceFilters as $resourceFilter) {
             if (null !== $value = $request->get($resourceFilter['name'])) {
                 $resourceFilter['value'] = $value;
                 $filters[] = $resourceFilter;
             }
         }
 
+        // Add order filters
+        $requestOrderFilters = $request->get('order');
+        foreach ($requestOrderFilters as $key => $value) {
+            // @TODO change of whitelist, currently is the one of the where filter
+            foreach ($resourceFilters as $resourceFilter) {
+                if ($resourceFilter['name'] === $key
+                    && ('asc' === strtolower($value) || 'desc' === strtolower($value))) {
+                    $order[$key] = $value;
+                }
+            }
+        }
+
+        // If no order found take the default
+        if (0 === count($order)) {
```

Comment:
@sroze this a convenient to be able to sort all API returns by id DESC (the most recent entries first) and this is a very common need and we already use this feature on one of our private project.
But you're right for the performance impact. In the current implementation, this feature is disabled by default: https://github.com/dunglas/DunglasJsonLdApiBundle/blob/master/DependencyInjection/Configuration.php#L42

It must stay disabled by default.